pkg/util/installinfo/install_info.go

```diff
@@ -166,3 +247,92 @@ func logVersionHistoryToFile(versionHistoryFilePath, installInfoFilePath, agentV
 		return
 	}
 }
+
+// HandleSetInstallInfo is an HTTP handler for setting install info at runtime
+func HandleSetInstallInfo(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost && r.Method != http.MethodPut {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	var req SetInstallInfoRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		respondWithError(w, http.StatusBadRequest, "Invalid JSON payload: "+err.Error())
+		return
+	}
+
+	installInfo := &InstallInfo{
+		Tool:             req.Tool,
+		ToolVersion:      req.ToolVersion,
+		InstallerVersion: req.InstallerVersion,
+	}
+
+	if err := SetRuntimeInstallInfo(installInfo); err != nil {
+		respondWithError(w, http.StatusBadRequest, "Failed to set install info: "+err.Error())
+		return
+	}
+
+	respondWithSuccess(w, "Install info set successfully")
+}
+
+// HandleGetInstallInfo is an HTTP handler for getting current install info
+func HandleGetInstallInfo(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	installInfo, err := Get(pkgconfigsetup.Datadog())
+	if err != nil {
+		respondWithError(w, http.StatusInternalServerError, "Failed to get install info: "+err.Error())
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(installInfo); err != nil {
+		log.Errorf("Failed to encode install info response: %v", err)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+}
+
+// HandleClearInstallInfo is an HTTP handler for clearing runtime install info override
+func HandleClearInstallInfo(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodDelete {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
```

Comment:
Same comment about not needing this extract check if we are using the mux filters